apis/v1alpha2/policy_types.go

```diff
@@ -100,6 +100,56 @@ type LocalPolicyTargetReferenceWithSectionName struct {
 	SectionName *SectionName `json:"sectionName,omitempty"`
 }
 
+// PolicyTargetReferenceWithSectionName identifies an API object to apply a
+// direct policy to. This should be used as part of Policy resources that can
+// target single resources. For more information on how this policy attachment
+// mode works, and a sample Policy resource, refer to the policy attachment
+// documentation for Gateway API.
+//
+// If the specified object namespace is in a different namespace from the policy itself,
```

Comment:
Add a section that describes the use case for this type, i.e., "This should only be used when a namespace is required for identifying the proper target."